traitlets/traitlets.py

```diff
@@ -499,18 +499,33 @@ def init_default_value(self, obj):
         obj._trait_values[self.name] = value
         return value
 
+    def get_env_value(self):
+        """
+        Gets the value of any environment-variable named as `env` metadata.
+
+        :return:
+            the textual value of the environment variable, or None, meaning
+            that no `envvar` metadata is defined.
+        """
+        env_var = self.metadata.get('envvar')
+        return env_var and os.environ.get(env_var)
```

Comment:
In general, any environment variable defined and empty should behave the same as that environment variable being undefined. It's probably appropriate to return Undefined in this case.